.dotnet/src/Custom/Batch/BatchClient.Protocol.cs

```diff
@@ -10,33 +11,72 @@ public partial class BatchClient
     /// <summary>
     /// [Protocol Method] Creates and executes a batch from an uploaded file of requests
     /// </summary>
+    /// <param name="returnWhen"> <see cref="ReturnWhen.Completed"/> if the
+    /// method should return when the service has finished running the 
+    /// operation, or <see cref="ReturnWhen.Started"/> if it should return 
+    /// after the operation has been created but may not have completed 
+    /// processing. </param>
     /// <param name="content"> The content to send as the body of the request. </param>
     /// <param name="options"> The request options, which can override default behaviors of the client pipeline on a per-call basis. </param>
     /// <exception cref="ArgumentNullException"> <paramref name="content"/> is null. </exception>
     /// <exception cref="ClientResultException"> Service returned a non-success status code. </exception>
-    /// <returns> The response returned from the service. </returns>
-    public virtual async Task<ClientResult> CreateBatchAsync(BinaryContent content, RequestOptions options = null)
+    /// <returns> A <see cref="BatchOperation"/> that can be used to wait for 
+    /// the operation to complete, or cancel the operation. </returns>
+    public virtual async Task<BatchOperation> CreateBatchAsync(ReturnWhen returnWhen, BinaryContent content, RequestOptions options = null)
     {
         Argument.AssertNotNull(content, nameof(content));
 
         using PipelineMessage message = CreateCreateBatchRequest(content, options);
-        return ClientResult.FromResponse(await _pipeline.ProcessMessageAsync(message, options).ConfigureAwait(false));
+
+        PipelineResponse response = await _pipeline.ProcessMessageAsync(message, options).ConfigureAwait(false);
+
+        using JsonDocument doc = JsonDocument.Parse(response.Content);
```

Comment:
That is fair.  If you feel strongly, it might not be too hard to update this to use Utf8JsonReader instead, but I'd have to look -- not sure how complex that would be offhand.

Reply:
I think we should add APIs (to BCL or SystemClientModel) based on https://www.rfc-editor.org/rfc/rfc6901 instead of using Utf8Reader directly. 
The usage would be:
```csharp
string batchId = JsonPointer.ReadString(binaryData, "batchId"u8);
```

Reply:
Saving this to come back to in a separate PR